xds/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,64 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package bootstrap provides the functionality to register possible options
+// for aspects of the xDS client through the bootstrap file.
+//
+// Experimental
+//
+// Notice: This package is EXPERIMENTAL and may be changed or removed
+// in a later release.
+package bootstrap
+
+import (
+	"encoding/json"
+
+	"google.golang.org/grpc/credentials"
+)
+
+// registry is a map from credential type name to Credential builder.
+var registry = make(map[string]Credentials)
+
+// Credentials interface encapsulates a credential provider
+// that can be used for communicating with the xDS Management server.
+type Credentials interface {
+	// Build returns a credential bundle associated with the Builder.
+	Build(config json.RawMessage) (credentials.Bundle, error)
+	// Name returns the credential name associated with this credential.
+	Name() string
+}
+
+// RegisterCredentials registers the credential builder that can be used to communicate
```

Comment:
Nit: TIL that credential is also a plural form. And sometimes it feels odd to use `credentials` since we think we are not referring to a plural form. I find using `creds` in comments simplifies things instead of having to think about `credentials` vs `credential`.

Reply:
"Credential" is plural?  What?  Source?  _"Credentials"_ is singular for us, though, since it's a type.

How about something like `RegisterCredentials registers a Credentials used for connecting to the xds management server.`

Reply:
> "Credential" is plural? What? Source? "Credentials" is singular for us, though, since it's a type.

I think I read it wrong, was reading too quickly:
https://www.merriam-webster.com/dictionary/credential